homeassistant/components/suez_water/__init__.py

```diff
@@ -11,30 +13,42 @@
 from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
 
 from .const import CONF_COUNTER_ID, DOMAIN
+from .coordinator import SuezWaterCoordinator
 
 PLATFORMS: list[Platform] = [Platform.SENSOR]
 
 
+def _get_client(entry: ConfigEntry) -> SuezClient:
+    try:
+        client = SuezClient(
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+            entry.data[CONF_COUNTER_ID],
+            provider=None,
+        )
+        if not client.check_credentials():
+            raise ConfigEntryError
+    except PySuezError as ex:
+        raise ConfigEntryNotReady from ex
+    return client
+
+
+async def _get_coordinator(
+    hass: HomeAssistant, entry: ConfigEntry
+) -> SuezWaterCoordinator:
+    loop = asyncio.get_running_loop()
+    client = await loop.run_in_executor(None, _get_client, entry)
+    coordinator = SuezWaterCoordinator(hass, client, entry.data[CONF_COUNTER_ID])
+    await coordinator.async_config_entry_first_refresh()
+    return coordinator
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Suez Water from a config entry."""
 
-    def get_client() -> SuezClient:
-        try:
-            client = SuezClient(
-                entry.data[CONF_USERNAME],
-                entry.data[CONF_PASSWORD],
-                entry.data[CONF_COUNTER_ID],
-                provider=None,
-            )
-            if not client.check_credentials():
-                raise ConfigEntryError
-        except PySuezError as ex:
-            raise ConfigEntryNotReady from ex
-        return client
-
-    hass.data.setdefault(DOMAIN, {})[
-        entry.entry_id
-    ] = await hass.async_add_executor_job(get_client)
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = await _get_coordinator(
+        hass, entry
+    )
```

Comment:
No need to create a separate function. Please use `await hass.async_add_executor_job(...)`